src/port1.0/portbump.tcl

```diff
@@ -0,0 +1,315 @@
+        # no longer has the file served its error page with a 200 response.
+        ui_notice "***"
+        ui_notice "The non-matching file appears to be HTML. See this page for possible reasons"
+        ui_notice "for the bump mismatch:"
+        ui_notice "<https://trac.macports.org/wiki/MisbehavingServers>"
+        ui_notice "***"
+        ui_notice "The file has been moved to: $htmlfile_path"
+        
+        return -code error "[msgcat::mc "Unable to verify file checksums"]"
+    } else {
+        # Show the desired checksum line for easy cut-paste
+        # based on the previously calculated values, plus our default types
+        set sums {}
+
+        foreach distfile $all_dist_files {
+            if {[llength $all_dist_files] > 1} {
+                lappend sums $distfile
+            }
+
+            set missing_types $portchecksum::default_checksum_types
+
+            # Append the string for the calculated types and note any of
+            # our default types that were already calculated
+            if {[info exists calculated_checksums_array($distfile)] && [llength $calculated_checksums_array($distfile)]} {
+                set calculated_checksums $calculated_checksums_array($distfile)
+                foreach {type sum} $calculated_checksums {
+                    set found [lsearch -exact ${missing_types} ${type}];
+                    if { ${found} != -1} {
+                        set missing_types [lreplace ${missing_types} ${found} ${found}]
+                    }
+                }
+            }
+
+            # Append the string for any of our default types that were
+            # not previously calculated
+            if {[llength $missing_types]} {
+                # get the full path of the distfile.
+                set fullpath [file join $distpath $distfile]
+                if {![file isfile $fullpath]} {
+                    return -code error "$distfile does not exist in $distpath"
+                }
+                # foreach type $missing_types {
+                #     lappend both_checksums $type "-" [portchecksum::calc_$type $fullpath]
+                # }
+            }
+        }
+
+        global version revision
+
+        ui_msg "We will bump these:"
+        foreach {type sum calculated_sum} $both_checksums {
+            ui_msg [format "Old %-8s %s" ${type}: $sum]
+            ui_msg [format "New %-8s %s" ${type}: $calculated_sum]
+        }
+
+        if {[info exists macports::ui_options(questions_yesno)]} {
+            set retval [$macports::ui_options(questions_yesno) $msg "BumpPrompt" "" {y} 0 "Do you approve of these changes?"]
+            if {$retval == 0} {
+                # User said yes
+                # TODO: Ask to reset revision number only if > 0 and pass revision number to message.
+                set retval [$macports::ui_options(questions_yesno) $msg "BumpPrompt" "" {y} 0 [format "Do you want to reset revision number (currently %s) to 0?" $revision]]
```

Comment:
Same here, this part contains unimplemented TODOs. If we can move those to a commit on top, I'd merge the changes below.